### PR TITLE
1889 - changelog for v0.5.0

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

part of #1889

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

**Review by rijobro — APPROVED** [review]
Very comprehensive, thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+## [0.5.0] - 2020-04-09
+### Added
+* Overview document for [feature highlights in v0.5.0](https://github.com/Project-MONAI/MONAI/blob/master/docs/source/highlights.md)
+* Invertible spatial transforms
```
Comment (Nic-Ma) [inline CHANGELOG.md:11]: Added TransformInverter handler.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+  * `PadListDataCollate` with inverse support
+* Support of slicing syntax for `Dataset`
+* Initial Torchscript support for the loss modules
+* Learning rate finder
+* Allow for missing keys in the dictionary-based transforms
+* Various summary and plotting utilities for Jupyter notebooks
+* Contributor Covenant Code of Conduct
+* Major CI/CD enhancements covering the tutorial repository
+* Fully compatible with PyTorch 1.8
+* Initial nightly CI/CD pipelines using Nvidia Blossom Infrastructure
```
Comment (Nic-Ma) [inline CHANGELOG.md:47]: Added support to load checkpoint with mismatch names and shapes for transfer learning.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+### Changed
+* Enhanced `list_data_collate` error handling
+* Unified iteration metric APIs
+* `densenet*` extensions are renamed to `DenseNet*`
+* `se_res*` network extensions are renamed to `SERes*`
+* Transform base APIs are rearranged into `compose`, `inverse`, and `transform`
+* `_do_transform` flag for the random augmentations is unified via `RandomizableTransform`
+* Decoupled post-processing steps, e.g. `softmax`, `to_onehot_y`, from the metrics computations
+* Moved the distributed samplers to `monai.data.samplers` from `monai.data.utils`
+* Engine's data loaders now accept generic iterables as input
```
Comment (Nic-Ma) [inline CHANGELOG.md:58]: workflows accepts additional custom events and state properties.

**Pull request merged**